SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -146,225 +146,284 @@ void ComponentReader::ApplyComponentUpdate(const Worker_ComponentId ComponentId,
+	const FSpatialConditionMapFilter ConditionMap(&Channel, bIsClient);
 
+	for (uint32 FieldId : ApplySchemaObjectData.UpdatedIds)
 	{
-		// Scoped to exclude OnRep callbacks which are already tracked per OnRep function
-		SCOPE_CYCLE_COUNTER(STAT_ReaderApplyPropertyUpdates);
-
-		Worker_EntityId EntityId = Channel.GetEntityId();
-		TArray<FSpatialGDKSpanId> CauseSpanIds;
-		if (bEventTracerEnabled)
+		// FieldId is the same as rep handle
+		if (FieldId == 0 || (int)FieldId - 1 >= BaseHandleToCmdIndex.Num())
 		{
-			CauseSpanIds = EventTracer->GetAndConsumeSpansForComponent(EntityComponentId(EntityId, ComponentId));
+			UE_LOG(LogSpatialComponentReader, Error,
+				   TEXT("ApplySchemaObject: Encountered an invalid field Id while applying schema. Object: %s, Field: %d, Entity: "
+						"%lld, Component: %d"),
+				   *Object.GetPathName(), FieldId, Channel.GetEntityId(), ComponentId);
+			continue;
 		}
 
-		for (uint32 FieldId : UpdatedIds)
+		int32 CmdIndex = BaseHandleToCmdIndex[FieldId - 1].CmdIndex;
+		const FRepLayoutCmd& Cmd = Cmds[CmdIndex];
+		const FRepParentCmd& Parent = Parents[Cmd.ParentIndex];
+		const int32 ShadowOffset = Cmd.ShadowOffset;
+
+		// Special Case properties are properties that are used in building the FSpatialConditionMapFilter,
+		// and the client must read them before rebuilding the condition map for all other properties.
+		bool bIsSpecialCaseProperty = false;
+		if (ApplySchemaObjectData.ProcessFieldType != ProcessFieldType_All)
 		{
-			// FieldId is the same as rep handle
-			if (FieldId == 0 || (int)FieldId - 1 >= BaseHandleToCmdIndex.Num())
+			const bool bProcessOnlySpecialCases = ApplySchemaObjectData.ProcessFieldType == ProcessFieldType_OnlySpecialCase;
+			auto ShouldConsiderProperty = [&Cmd, &bProcessOnlySpecialCases]() -> bool {
+				if (bProcessOnlySpecialCases)
+				{
+					return UNLIKELY((int32)AActor::ENetFields_Private::RemoteRole == Cmd.ParentIndex)
+						|| UNLIKELY((int32)AActor::ENetFields_Private::Role == Cmd.ParentIndex)
+						|| UNLIKELY((int32)AActor::ENetFields_Private::ReplicatedMovement == Cmd.ParentIndex);
+				}
+				// We still need to process the full ReplicatedMovement struct, because only its bRepPhysics member is a special case.
+				// So if we're processing AllButSpecialCase, we only have to skip Role and Remote Role
+				return !(UNLIKELY((int32)AActor::ENetFields_Private::RemoteRole == Cmd.ParentIndex)
+					|| UNLIKELY((int32)AActor::ENetFields_Private::Role == Cmd.ParentIndex));
+			};
+
+			const bool bShouldConsiderProperty = ShouldConsiderProperty();
+			if (!bShouldConsiderProperty)
 			{
-				UE_LOG(LogSpatialComponentReader, Error,
-					   TEXT("ApplySchemaObject: Encountered an invalid field Id while applying schema. Object: %s, Field: %d, Entity: "
-							"%lld, Component: %d"),
-					   *Object.GetPathName(), FieldId, Channel.GetEntityId(), ComponentId);
 				continue;
 			}
+			bIsSpecialCaseProperty = bShouldConsiderProperty && bProcessOnlySpecialCases;
```

Comment:
`bShouldConsiderProperty` will always be true here (since it would've hit `continue` in the if above if it was false)

Reply:
Thanks, fixed this up so hopefully it's more clear.